### PR TITLE
fix: Format the interfaces in map values as json

### PR DESCRIPTION
**Pull request by seshapad**
Signed-off-by: seshapad <seshapad@in.ibm.com>

### CODE REVIEW

File: common/utils.go [hunk 1]
```diff
@@ -1065,6 +1065,31 @@ func MarshalObjToYaml(obj runtime.Object) ([]byte, error) {
...
+		}
+		var b bytes.Buffer
+		encoder := json.NewEncoder(&b)
+		if err := encoder.Encode(value); err != nil {
+			logrus.Error("Unable to unmarshal data to json while converting map interfaces to string")
+			continue
+		}
+		strValue := string(b.Bytes())
+		strValue = strings.TrimSpace(strValue)
+		strValue = strings.Trim(strValue, `\r\n`)
```
Comment (ashokponkumar) [inline common/utils.go:1087]: Wont trimspace remove this too?

File: common/utils.go [hunk 2]
```diff
@@ -1065,6 +1065,31 @@ func MarshalObjToYaml(obj runtime.Object) ([]byte, error) {
...
 
+// StringifyMap stringifies the map values
+func StringifyMap(inputMap map[string]interface{}) map[string]interface{} {
+	for key, value := range inputMap {
+		if value == nil {
+			inputMap[key] = nil
+			continue
+		}
+		if reflect.TypeOf(value).Kind() == reflect.String {
+			inputMap[key] = value.(string)
```
Comment (ashokponkumar) [inline common/utils.go:1076]:
```suggestion
		if val, ok := value.(string); ok {
			inputMap[key] = val
```

File: common/utils.go [hunk 3]
```diff
@@ -1065,6 +1065,31 @@ func MarshalObjToYaml(obj runtime.Object) ([]byte, error) {
 	return b.Bytes(), nil
 }
 
+// StringifyMap stringifies the map values
+func StringifyMap(inputMap map[string]interface{}) map[string]interface{} {
+	for key, value := range inputMap {
+		if value == nil {
+			inputMap[key] = nil
```
Comment (ashokponkumar) [inline common/utils.go:1072]: do we need this assignment if it is already nil?

File: common/utils.go [hunk 4]
```diff
@@ -1065,6 +1065,31 @@ func MarshalObjToYaml(obj runtime.Object) ([]byte, error) {
 	return b.Bytes(), nil
 }
 
+// StringifyMap stringifies the map values
+func StringifyMap(inputMap map[string]interface{}) map[string]interface{} {
```
Comment (ashokponkumar) [inline common/utils.go:1069]: StringifyMap is a very specify function that is not a general usecase, so won't it be better to just place it in the collector or collect type package itself? or do we see this specific situation occurring in other places?

File: transformer/cloudfoundrytransformer.go [hunk 5]
```diff
@@ -321,6 +321,7 @@ func getCfAppInstance(path string, appname string) (collecttypes.CfApp, error) {
 	if err := common.ReadMove2KubeYaml(path, &cfinstanceappsfile); err != nil {
 		return collecttypes.CfApp{}, err
 	}
+	cfinstanceappsfile = collecttypes.FormatMapsWithInterface(cfinstanceappsfile)
```
Comment (ashokponkumar) [inline transformer/cloudfoundrytransformer.go:324]: Do we need to call this here? This will anyway be loaded as a string, isn't it?

**Review by ashokponkumar — CHANGES_REQUESTED** [review]
Nice. Few comments..

File: common/utils.go (hunk 3 above)
Comment (seshapad) [reply common/utils.go]: done

File: common/utils.go (hunk 1 above)
Comment (seshapad) [reply common/utils.go]: yes. made changes accordingly.

File: common/utils.go (hunk 4 above)
Comment (seshapad) [reply common/utils.go]: done.

File: transformer/cloudfoundrytransformer.go (hunk 5 above)
Comment (seshapad) [reply transformer/cloudfoundrytransformer.go]: done.

**Review by seshapad — COMMENTED** [review]
Signed-off-by: seshapad <seshapad@in.ibm.com>

File: types/collection/cfapps.go [hunk 6]
```diff
@@ -52,3 +57,42 @@ func NewCfApps() CfApps {
...
+		}
+		var b bytes.Buffer
+		encoder := json.NewEncoder(&b)
+		if err := encoder.Encode(value); err != nil {
+			logrus.Error("Unable to unmarshal data to json while converting map interfaces to string")
+			continue
+		}
+		strValue := string(b.Bytes())
+		strValue = strings.TrimSpace(strValue)
+		//strValue = strings.Trim(strValue, `\r\n`)
```
Comment (ashokponkumar) [inline types/collection/cfapps.go:94]: So we need this line?

**Review by ashokponkumar — CHANGES_REQUESTED** [review]
Minor comment. Also, did we check with pr with e2e app?

**Comment by seshapad** [comment]
> Minor comment. Also, did we check with pr with e2e app?

@ashokponkumar yes. It works!

**Review by ashokponkumar — APPROVED** [review]

**Pull request merged**